netket/optimizer/qgt/qgt_onthefly_logic.py

```diff
@@ -14,10 +14,29 @@
 
 import jax
 from jax.tree_util import Partial
-from functools import partial
+from functools import partial, wraps
 from netket.stats import subtract_mean
 from netket.utils import mpi
 from netket.jax import tree_conj, tree_axpy
+from netket.jax import (
+    scanmap,
+    scan_accum,
+    scan_append,
+    batch_args,
+    unbatch_output,
+    compose,
+)
+
+
+# temporary fix to make linear_call jittable
+# TODO remove once in jax
+from jax.interpreters import xla
+from jax._src.custom_derivatives import linear_call_p, _linear_call_impl
+
+xla.initial_style_translations[linear_call_p] = xla.lower_fun_initial_style(
+    _linear_call_impl
+)
+
```

Comment:
Jax just made a new release which includes your PR, so I'd remove this and you can bump the minimum jax version to 0.2.21

```suggestion
```